src/java.desktop/share/classes/sun/font/StrikeCache.java

```diff
@@ -136,109 +136,179 @@ private static VarHandle getVarHandle(StructLayout struct, String name) {
         return MethodHandles.insertCoordinates(h, 1, 0L).withInvokeExactBehavior();
     }
 
-    private static final VarHandle xAdvanceHandle = getVarHandle(GlyphImageLayout, "xAdvance");
-    private static final VarHandle yAdvanceHandle = getVarHandle(GlyphImageLayout, "yAdvance");
-    private static final VarHandle widthHandle    = getVarHandle(GlyphImageLayout, "width");
-    private static final VarHandle heightHandle   = getVarHandle(GlyphImageLayout, "height");
-    private static final VarHandle rowBytesHandle = getVarHandle(GlyphImageLayout, "rowBytes");
-    private static final VarHandle managedHandle  = getVarHandle(GlyphImageLayout, "managed");
-    private static final VarHandle topLeftXHandle = getVarHandle(GlyphImageLayout, "topLeftX");
-    private static final VarHandle topLeftYHandle = getVarHandle(GlyphImageLayout, "topLeftY");
-    private static final VarHandle cellInfoHandle = getVarHandle(GlyphImageLayout, "cellInfo");
-    private static final VarHandle imageHandle    = getVarHandle(GlyphImageLayout, "image");
+    private static VarHandle xAdvanceHandle;
+    private static VarHandle yAdvanceHandle;
+    private static VarHandle widthHandle;
+    private static VarHandle heightHandle;
+    private static VarHandle rowBytesHandle;
+    private static VarHandle managedHandle;
+    private static VarHandle topLeftXHandle;
+    private static VarHandle topLeftYHandle;
+    private static VarHandle cellInfoHandle;
+    private static VarHandle imageHandle;
+
+    private static VarHandle getXAdvanceHandle() {
```

Comment:
Quick question: is there no concern with multi-threading / race conditions for these lazy creation methods? (e.g. requiring double-checked locking or similar)

This looks like a good use case for `StableValue`, if it were ready!

Reply:
There won't be multiple threads except in some really unusual case so I don't think it is a problem